controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller_test.go

```diff
@@ -0,0 +1,165 @@
+package spaceprovisionerconfig
+
+import (
+	"context"
+	"errors"
+	"os"
+	"testing"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/apis"
+	"github.com/codeready-toolchain/toolchain-common/pkg/condition"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestSpaceProvisionerConfigValidation(t *testing.T) {
```

Comment:
could you please add a test that checks the updates of the conditions:
* is true as part of the setup -> changes to false because the ToolchainCluster is gone
* is false as part of the setup -> changes to true because the ToolchainCluster was created

Reply:
it's always better to test more things in unit tests than in e2e tests. 
As for the setup, what we do is that we create the object in the expected state (with the condition already set) as part of the "// given" section:
```go
spc := testSpc.NewSpaceProvisionerConfig("spc", test.HostOperatorNs, "cluster1", WithReadyCondition())
r, req, cl := prepareReconcile(t, spc, ...
```
or
```go
spc := testSpc.NewSpaceProvisionerConfig("spc", test.HostOperatorNs, "cluster1", WithUnreadyCondition("someReason", "some message"))
r, req, cl := prepareReconcile(t, spc, ...
```

in this way, you can easily verify the scenario in an isolated way. You have the CR in the given state -> what happens if under the given conditions during the reconcile.

Reply:
The unit tests are now implemented.